api/src/main/java/marquez/MarquezApp.java

```diff
@@ -154,6 +171,35 @@ public void run(@NonNull MarquezConfig config, @NonNull Environment env) {
     Exclusions.use(exclusions);
   }
 
+  private OpenSearchClient newOpenSearchClient(SearchConfig searchConfig) {
```

Comment:
We should encapsulate the `OpenSearchClient` instantiation in the constructor of `SearchService` by passing `SearchConfig`:

```
new SearchService(SearchConfig)
```

Reply:
💡 Great idea. This would move the init code out of this non relevant final.